### PR TITLE
Poisoned healthbar feedback

### LINKED ISSUE #638: [FEATURE] When poisoned, change the health bar color to green
## 🚀 Feature

We're currently using a smoke feedback to represent the poisoned status of a character.
It would be better to remove that feedback and just change the affected character's health bar color to green while the poison effect lasts.

### PR DESCRIPTION
**Pull request by ncontinanza**
Closes #638 

This PR removes the current feedback for the Poisoned state. Instead of that, we'll now simply change the poisoned character's health bar color to green while the effect lasts.

As we currently don't have any ability that applies the Poisoned effect on an enemy, we'll have to test this new feature indirectly.

A simple way to test this is to simply negate the condition for this feedback. That is, go to Battle.cs and change 
```
if (playerUpdate.Effects.ContainsKey((ulong)PlayerEffect.Poisoned))
```
 for 
```
if (!playerUpdate.Effects.ContainsKey((ulong)PlayerEffect.Poisoned))
``` 
By doing this, when you start a new game, your character's health bar will be green during the whole match. 

A more realistic way to test it is to give this behaviour to another effect. For example, you can replace the Paralyzed case in Battle.cs for the following code:

```
MMHealthBar healthBar = player.GetComponent<MMHealthBar>();

if (playerUpdate.Effects.ContainsKey((ulong)PlayerEffect.Paralyzed))
{
    characterSpeed = 0f;
    healthBar.ForegroundColor = GetHealthBarGradient(MMColors.Green);
}
else
{
    if (healthBar.ForegroundColor.Equals(GetHealthBarGradient(MMColors.Green)))
    {
        healthBar.ForegroundColor = GetHealthBarGradient(MMColors.BestRed);
    }
}
```
You'll also need to comment the code that is immediately below (the if-else that handles de Poisoned effect).
This way, when you paralyze another character with H4ck, its health bar will turn green as if it were poisoned.

### CODE REVIEW

File: client/Assets/Scripts/Battle.cs [hunk 1]
```diff
@@ -880,6 +878,20 @@ float characterSpeed
...
+        {
+            healthBar.ForegroundColor = GetHealthBarGradient(MMColors.Green);
+        }
+        else
+        {
+            if (healthBar.ForegroundColor.Equals(GetHealthBarGradient(MMColors.Green)))
+            {
+                healthBar.ForegroundColor = GetHealthBarGradient(MMColors.BestRed);
+            }
+        }
```
[the remark this answers is not in the document]
Comment (ncontinanza) [reply client/Assets/Scripts/Battle.cs]: Right!

File: client/Assets/Scripts/Battle.cs [hunk 2]
```diff
@@ -905,4 +917,21 @@ private bool PlayerIsAlive(Player playerUpdate)
...
+                new GradientColorKey(color, 0),
+                new GradientColorKey(color, 1f)
+            },
+            alphaKeys = new GradientAlphaKey[2]
+            {
+                new GradientAlphaKey(1, 0),
+                new GradientAlphaKey(1, 1)
+            }
+        };
+    }
```
[the remark this answers is not in the document]
Comment (ncontinanza) [reply client/Assets/Scripts/Battle.cs]: Absolutely

**Review by mgrunwaldt — APPROVED** [review]
Great work!

**Review by AngieDutra — CHANGES_REQUESTED** [review]
Check the overlay effects, damage feedback and death feedback of each character prefab. Also, make sure to checkout from main SkillRange and TerrainZone. Those should never be modified.

**Review by AngieDutra — APPROVED** [review]
Looking good! ✨

**Review by mgrunwaldt — APPROVED** [review]

**Pull request merged**